src/org/mozilla/javascript/TokenStream.java

```diff
@@ -1185,6 +1206,62 @@ final int getToken() throws IOException
         }
     }
 
+    /*
+     * Helper to read the next digits according to the base
+     * and ignore the number separator if there is one.
+     */
+    private int readDigits(int base, int c) throws IOException {
+        if ((base == 10 && isDigit(c))
+                || (base == 16 && isHexDigit(c))
+                || (base == 8 && isOctalDigit(c))
+                || (base == 2 && isDualDigit(c))) {
+            addToString(c);
+
+            c = getChar();
+            if (c == EOF_CHAR) {
+                return EOF_CHAR;
```

Comment:
FWIW according to "./gradlew jacocoTestReport" this is never hit in this function -- everything else is.

Reply:
I think this is because there is no test for this case - last char of a script has to be a digit. I dont like to add test to the 262 suite - should we write a separate test for this?

Reply:
Ok, just writing a test, give me 10 min.

Reply:
done